src/zc/buildout/easy_install.py

```diff
@@ -1619,25 +1622,19 @@ def unpack_egg(location, dest):
     setuptools.archive_util.unpack_archive(location, dest)
 
 
-WHEEL_TO_EGG_WARNING = """
-Using unpack_wheel() shim over the deprecated wheel_to_egg() hook.
-Please update your wheel extension implementation for one that installs a .whl
-handler in %s.UNPACKERS
-""".strip() % (__name__,)
+WHEEL_WARNING = """
+*.whl file detected (%s), you'll need setuptools > 38.2.3 for that
+or an extension like buildout.wheel
+"""
+
 
 def unpack_wheel(location, dest):
-    # Deprecated backward compatibility shim. Please do not use.
-    logger.warning(WHEEL_TO_EGG_WARNING)
-    basename = os.path.basename(location)
-    dists = setuptools.package_index.distros_for_location(location, basename)
-    # `wheel_to_egg()` might generate zipped eggs, so we have to make sure we
-    # get unpacked eggs in the end:
-    tmp_dest = tempfile.mkdtemp(dir=dest)
-    wheel_to_egg(list(dists)[0], tmp_dest)
-    [egg] = glob.glob(os.path.join(tmp_dest, '*.egg'))
-    unpack_egg(egg, dest)
-    shutil.rmtree(tmp_dest)
-    
+    if SETUPTOOLS_SUPPORTS_WHEELS:
+        wheel = Wheel(location)
+        wheel.install_as_egg(os.path.join(dest, wheel.egg_name()))
+    else:
+        logger.warning(WHEEL_WARNING, location)
```

Comment:
> I don't like putting in the old shim as it was quite a horrid piece of code that only did anything meaningful if you had an old version of buildout.wheel. In that case you're probably also using an older buildout, I'd say (or you need to upgrade). So I'd rather keep the shim out.

You're right.

Still, this corner case in the else clause means something that is guaranteed to cause an error condition later on that is harder to track to the cause (user has old `setuptools`, with old `buildout.wheel`).

And `wheel_to_egg()` is still dead code, so would you mind turning this warning into an error and removing `wheel_to_egg()`?

Reply:
`wheel_to_egg()` is gone

Reply:
And the warning is an error now.